### PR TITLE
Prefer TypeConverter over TryParse

### LINKED ISSUE #46384: DateOnly custom TypeConverter is ignored when using FromQuery
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Since .NET 7 and the new support for `DateOnly`, our custom `TypeConverter ` for `DateOnly` is not invoked anymore when the controller parameter is an object initialized with `[FromQuery]` on a GET request

Our client sends the date with time at zero (YYYY-MM-DDT00:00:00Z), so our converter is taking the date part and ignoring the time, which worked well in .NET 6.
In .NET 7, with exactly the same code, we have a validation error `The value '2023-01-31T00:00:00Z' is not valid.` because the new converter is taking over, and our custom is not invoked.

I'm not sure whether this is a bug or us not using it properly, but it worked previously...
A workaround that don't imply changing the contract is welcome.

### Expected Behavior

No behavior change.
The default TypeConverter is ignored, and our custom one is invoked.

### Steps To Reproduce

The demo are the default .NET Web API template (weather forecast), which I added a simplified version of our custom TypeConverter, a global registration of it, and an object parameter to the forecast endpoint, which contains a DateOnly.

WebApplication1 is .NET 7 => with `2023-01-31T00:00:00Z` as the date parameter => TypeConverter is not invoked, validation error
WebApplication2 is .NET 6 => with `2023-01-31T00:00:00Z` => TypeConverter is invoked, works

https://github.com/killergege/net_typeconverter_demo

### Exceptions (if any)

_No response_

### .NET Version

7.0.102

### Anything else?

Visual Studio 2022

```
.NET SDK:
 Version:   7.0.102
 Commit:    4bbdd14480

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.102\

Host:
  Version:      7.0.2
  Architecture: x64
  Commit:       d037e070eb

.NET SDKs installed:
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.425 [C:\Program Files\dotnet\sdk]
  6.0.404 [C:\Program Files\dotnet\sdk]
  7.0.101 [C:\Program Files\dotnet\sdk]
  7.0.102 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

### PR DESCRIPTION
**Pull request by brunolins16**
Fix #46384

### CODE REVIEW

**Comment by brunolins16** [comment]
@dotnet/minimal-apis could you please review? 🙏

File: src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinderProvider.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public class SimpleTypeModelBinderProvider : IModelBinderProvider
     {
         ArgumentNullException.ThrowIfNull(context);
 
-        if (!context.Metadata.IsComplexType)
+        if (context.Metadata.IsConvertibleType)
```
Comment (captainsafia) [inline src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinderProvider.cs:21]: I assume that if `IsConvertibleType` is true, then we know that the model is not correlated with a complex type.

File: src/Mvc/test/Mvc.IntegrationTests/SimpleTypeModelBinderIntegrationTest.cs [hunk 2]
```diff
@@ -699,6 +702,50 @@ public async Task BindParameter_FromFormData_BindsCorrectly(Dictionary<string, S
         Assert.Equal(new[] { "line 1", "line 2" }, entry.RawValue);
     }
 
+    [Fact]
+    public async Task BindParameter_PrefersTypeConverter_OverTryParse()
```
Comment (captainsafia) [inline src/Mvc/test/Mvc.IntegrationTests/SimpleTypeModelBinderIntegrationTest.cs:706]: Is it helpful to add a test that verifies the `TryParse` will get called for types that don't have a converter?

**Review by captainsafia — APPROVED** [review]
Looks good overall! I left some non-blocking questions inline.

File: src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinderProvider.cs (hunk 1 above)
Comment (brunolins16) [reply src/Mvc/Mvc.Core/src/ModelBinding/Binders/SimpleTypeModelBinderProvider.cs:21]: Exactly:

https://github.com/dotnet/aspnetcore/blob/9b84c4b982e05260077e7759f780bb0ed0be554e/src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs#L445

**Pull request merged**